docs/design/2022-11-24-resource-manage-for-ddl.md

```diff
@@ -0,0 +1,148 @@
+# Proposal: resource manage for DDL
+- Author(s): [hawkingrei](https://github.com/hawkingrei) 
+- Tracking issue: #38025
+
+## Table of Contents
+
+<!-- TOC -->
+* [Abstract](#abstract)
+* [Background](#background)
+* [Design](#design)
+  * [Component](#component)
+  * [Dynamic goroutine pool](#dynamic-goroutine-pool)
+  * [Resource manager](#resource-manager)
+    * [scheduler interface](#scheduler-interface)
+    * [simple CPU utilization-based scheduler](#simple-CPU-utilization-based-scheduler)
+    * [Task scheduler](#task-scheduler)
+      * [Task meta](#task-meta)
+      * [Task meta management](#task-meta-management)
+      * [Task schedule](#task-schedule)
+
+<!-- TOC -->
+
+## Abstract
+
+This proposes a design of how to control the usage for CPU when to do DDL operator and we want to avoid concurrent ddl 
+affecting queries.
+
+## Background
+
+Currently, TiDB DDL is not well controlled, which may cause high CPU usage when we do DDL operations. For example, we 
+have a table with 1000 columns, and we add a column to it. It will cause high CPU usage. Concurrent DDL operations may 
```

Comment:
Currently, add a column is general DDL, it only change table meta data,  will not cause high CPU usage.